webroot/src/components/App/App.ts

```diff
@@ -88,12 +84,57 @@ class App extends Vue {
     //
     // Methods
     //
-    async setCurrentCompact() {
-        const compact = CompactSerializer.fromServer({ type: CompactType.ASLP }); // Temp until server endpoints define the user's compact
+    async handleAuth() {
+        const authType = this.setAuthType();
+
+        if (authType !== AuthTypes.PUBLIC) {
+            this.$store.dispatch('user/startRefreshTokenTimer', authType);
+            await this.getAccount();
+            await this.setCurrentCompact();
+        }
+    }
 
-        await this.$store.dispatch('user/setCurrentCompact', compact);
+    setAuthType() {
+        let authType: AuthTypes;
 
-        return compact;
+        if (authStorage.getItem(tokens?.staff?.AUTH_TOKEN)) {
+            authType = AuthTypes.STAFF;
+        } else if (authStorage.getItem(tokens?.licensee?.AUTH_TOKEN)) {
+            authType = AuthTypes.LICENSEE;
+        } else {
+            authType = AuthTypes.PUBLIC;
+        }
+
+        this.$store.dispatch('setAuthType', authType);
+
+        return authType;
+    }
+
+    async getAccount(): Promise<void> {
+        const { authType } = this.globalStore;
+
+        if (authType === AuthTypes.STAFF) {
+            await this.$store.dispatch('user/getStaffAccountRequest');
+        }
+    }
+
+    async setCurrentCompact(): Promise<void> {
+        const { currentCompact, model: user } = this.userStore;
+        const { permissions = [] } = user || {};
+        const userDefaultCompact = permissions?.[0]?.compact || null;
+
+        // If a current compact is not set or the current compact is not part of the user permissions
+        if (!currentCompact || !permissions.some((permission) => permission.compact.type === currentCompact.type)) {
+            await this.$store.dispatch('user/setCurrentCompact', userDefaultCompact);
+
+            // If the current route is not matching the newly set compact, the redirect
```

Comment:
as nit as you can get...

```suggestion
            // If the current route is not matching the newly set compact, then redirect
```

Reply:
Good catch!